app/main.py

```diff
@@ -1,2 +1,126 @@
+from typing import Any, Iterator, Hashable
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+    hash_value: int | float
+    next_node: "Node" = None
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self._length = 0
+        self._capacity = 8
+        self._hash_table = [None] * self._capacity
+
+    def _get_index_and_hash(self, key: Hashable) -> tuple:
+        keys_hash = hash(key)
+        index = keys_hash % self._capacity
+        return index, keys_hash
+
+    def _resize(self) -> None:
+        self._length = 0
+        self._capacity *= 2
+        old_hash_table = self._hash_table
+        self._hash_table = [None] * self._capacity
+
+        for node in old_hash_table:
+            while node is not None:
+                self[node.key] = node.value
+                node = node.next_node
```

Comment:
You can use `__setitem__` here

Reply:
Sorry @olena-hul i don't really understand . I'm already using `__setitem__` here `` self[node.key]` = node.value.` No?

Reply:
You can use `self[node.key]`, it is ok.